vllm/worker/model_runner.py

```diff
@@ -449,11 +471,25 @@ def _prepare_model_input(
                         raise ValueError(
                             "Multi-modal inputs are only supported by "
                             "vision language models.")
-
                     mm_kwargs = self.multi_modal_input_processor(mm_data)
                     for k, v in mm_kwargs.items():
                         multi_modal_kwargs_list[k].append(v)
 
+                if prompt_adapter_id > 0:
+                    prompt_adapter_requests.add(
+                        seq_group_metadata.prompt_adapter_request)
+
+                num_tokens = seq_group_metadata.\
+                                        prompt_adapter_num_virtual_tokens
+                pm = [prompt_adapter_id
+                      ] * num_tokens + [0] * (query_len - num_tokens)
+                prompt_adapter_index_mapping += pm
+                prompt_adapter_prompt_mapping.extend(
+                    [prompt_adapter_id] *
+                    (query_len if seq_group_metadata.sampling_params
+                     and seq_group_metadata.sampling_params.prompt_logprobs
+                     else 1))
```

Comment:
let's put this in a separate method for cleanliness

Reply:
sounds good

Reply:
On this feedback point, it didn't feel intuitive to put just this code in a separate function, but keep the lora code just above this(which looks almost similar) inline. Also this code extends the variables in code, doing that in another function would be pretty wieldy. What do you think?

Reply:
ok let's keep it in here for now